### PR TITLE
Room list badges

### LINKED ISSUE #2079: Iterate on room list indicators
Update the way pictos are displayed on the room list, following those rules from [Figma](https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?type=design&node-id=15647%3A53824&mode=design&t=4yTuk3MX1AIsAoy0-1):

image

So it depends on the fields
- `RoomInfo.userDefinedNotificationMode`
- `RoomInfo.numUnreadMessages` for the notification count (from this [PR](https://github.com/matrix-org/matrix-rust-sdk/pull/2953))
- `RoomInfo.numUnreadMentions` for the highlight count (from this [PR](https://github.com/matrix-org/matrix-rust-sdk/pull/2953))

### PR DESCRIPTION
**Pull request by bmarty**
Follow new rules defined in #2079 to display room badges in the room list.

Closes #2079

### CODE REVIEW

File: libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/roomlist/RoomSummary.kt [hunk 1]
```diff
@@ -36,13 +36,15 @@ sealed interface RoomSummary {
...
-    val avatarURLString: String?,
+    val avatarUrl: String?,
     val lastMessage: RoomMessage?,
-    val lastMessageTimestamp: Long?,
-    val unreadNotificationCount: Int,
-    val inviter: RoomMember? = null,
-    val notificationMode: RoomNotificationMode? = null,
-    val hasOngoingCall: Boolean = false,
-)
+    val numUnreadMessages: Int,
```
Comment (ganfra) [inline libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/roomlist/RoomSummary.kt:43]: I'd prefer a clear `numberOfUnreadMessages` and `numberOfUnreadMontions` name, but up to you

File: features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/components/RoomSummaryRow.kt [hunk 2]
```diff
@@ -173,40 +174,65 @@ private fun RowScope.LastMessageAndIndicatorRow(room: RoomListRoomSummary) {
...
+            modifier = Modifier.size(16.dp),
+            imageVector = CompoundIcons.VideoCallSolid,
+            contentDescription = null,
+            tint = ElementTheme.colors.unreadIndicator,
+        )
+    }
+}
+
+@Composable
+private fun RowScope.NotificationIcons(room: RoomListRoomSummary) {
```
Comment (ganfra) [inline features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/components/RoomSummaryRow.kt:196]: Pass only the 3 needed values?

File: features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/components/RoomSummaryRow.kt [hunk 3]
```diff
@@ -173,40 +174,65 @@ private fun RowScope.LastMessageAndIndicatorRow(room: RoomListRoomSummary) {
...
-        null, RoomNotificationMode.ALL_MESSAGES -> return
-        RoomNotificationMode.MENTIONS_AND_KEYWORDS_ONLY ->
-            Icon(
-                modifier = Modifier.size(16.dp),
-                contentDescription = null,
-                imageVector = CompoundIcons.Mention,
-                tint = tint,
-            )
-        RoomNotificationMode.MUTE ->
+private fun OnGoingCallIcon(room: RoomListRoomSummary) {
```
Comment (ganfra) [inline features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/components/RoomSummaryRow.kt:184]: Pass only the boolean

**Review by ganfra — APPROVED** [review]
Small remarks otherwise LGTM

File: libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/roomlist/RoomSummary.kt (hunk 1 above)
Comment (bmarty) [reply libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/roomlist/RoomSummary.kt:43]: Yes, always the same debate if we want to stick with the SDK name...

**Comment by bmarty** [comment]
(ignoring failing test for now, passing locally)

**Pull request merged**